app/controllers/admin_routes/manage_departments.py

```diff
@@ -29,18 +32,63 @@ def manage_departments():
                 return render_template('errors/403.html'), 403
 
         departmentTracy = Tracy().getDepartments()
+        allSupervisors= Supervisor.select().order_by(Supervisor.LAST_NAME)
         for dept in departmentTracy:
             d, created = Department.get_or_create(DEPT_NAME = dept.DEPT_NAME, ACCOUNT=dept.ACCOUNT, ORG = dept.ORG)
             d.save()
         department = Department.select()
         return render_template( 'admin/manageDepartments.html',
                                 title = ("Manage Departments"),
-                                department = department
+                                department = department,
+                                allSupervisors = allSupervisors
                                 )
     except Exception as e:
         print("Error Loading all Departments", e)
         return render_template('errors/500.html'), 500
 
+@admin.route("/admin/manageDepartments/<currentDepartment>", methods=['GET'])
+def getSupervisorsinDepartment(currentDepartment):
+    try:
+        currentUser = require_login()
+        if not currentUser:                    # Not logged in
+            return render_template('errors/403.html')
+        if not currentUser.isLaborAdmin:       # Not an admin
+            if currentUser.student: # logged in as a student
+                return redirect('/laborHistory/' + currentUser.student.ID)
+            elif currentUser.supervisor:
+                return render_template('errors/403.html'), 403
+        
+        supervisors = getSupervisorsForDepartment(currentDepartment)
+        currentDepartment=Department.get_by_id(currentDepartment)
+        supervisors= [model_to_dict(supervisor) for supervisor in supervisors]
+        return jsonify([model_to_dict(currentDepartment), supervisors])
+    except Exception as e:
+        return render_template('errors/500.html'), 500
+    
+@admin.route('/admin/manageDepartments/removeSupervisorFromDepartment', methods=['POST'])
+def removeSupervisorFromDepartment():
+    supervisorDeptRecord=request.form
+    supervisorDeptRecord = SupervisorDepartment.get_or_none(supervisor = supervisorDeptRecord['supervisor'], department = supervisorDeptRecord['department'])
```

Comment:
We should probably move the request.form and supervisor get to after we verify the user below.